channelserver/src/server.rs

```diff
@@ -0,0 +1,257 @@
+//! `ChannelServer` is an actor. It maintains list of connection client session.
+//! And manages available channels. Peers send messages to other peers in same
+//! channel through `ChannelServer`.
+
+// use std::sync::{Arc, Mutex};
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::time::Instant;
+
+use actix::prelude::{Actor, Context, Handler, Recipient};
+use rand::{self, Rng, ThreadRng};
+use uuid::Uuid;
+
+use logging::MozLogger;
+use perror;
+use settings::Settings;
+
+//pub type ChannelCollection = Arc<Mutex<HashMap<Uuid, HashMap<usize, Channel>>>>;
+//pub type SessionCollection = Arc<Mutex<HashMap<usize,Recipient<TextMessage>>>>;
+
+/// Chat server sends this messages to session
+#[derive(Message)]
+pub struct TextMessage(pub String);
+
+/// Message for chat server communications
+
+/// New chat session is created
+#[derive(Message)]
+#[rtype(usize)]
+pub struct Connect {
+    pub addr: Recipient<TextMessage>,
+    pub channel: Uuid,
+}
+
+/// Session is disconnected
+#[derive(Message)]
+pub struct Disconnect {
+    pub channel: Uuid,
+    pub id: usize,
+}
+
+/// Send message to specific channel
+#[derive(Message)]
+pub struct ClientMessage {
+    /// Id of the client session
+    pub id: usize,
+    /// Peer message
+    pub msg: String,
+    /// channel name
+    pub channel: Uuid,
+}
+
+#[derive(Hash, Eq, PartialEq, Clone, Debug)]
+pub struct Channel {
+    pub id: usize,
+    pub started: Instant,
+    pub msg_count: u8,
+    pub data_exchanged: usize,
+}
+
+/// `ChannelServer` manages chat channels and responsible for coordinating chat
+/// session. implementation is super primitive
```

Comment:
It took me a little while to figure out how `channels` and `sessions` relate to each other here.  IIUC a "session" is an individual client connected via websocket, while a "channel" is a set of such sessions, along with some metadata about their activity.  Is that accurate?  We should probably capture that in a comment here for posterity.

Reply:
Agreed. Heh, the original code was worse about that if it's any help, but I'll try to spell things out much better.